baseapp/abci.go

```diff
@@ -991,6 +995,13 @@ func (app *BaseApp) Commit() (*abci.CommitResponse, error) {
 		for _, abciListener := range abciListeners {
 			if err := abciListener.ListenCommit(ctx, *resp, changeSet); err != nil {
 				app.logger.Error("Commit listening hook failed", "height", blockHeight, "err", err)
+				if app.streamingManager.StopNodeOnErr {
+					rollbackErr := app.cms.RollbackToVersion(blockHeight - 1)
+					if rollbackErr != nil {
+						return nil, rollbackErr
```

Comment:
need, let's errors.Join both to not lose context on why we exited at the first place:

```go
return nil, errors.Join(fmt.Errorf("Commit listening hook failed: %w", err), rollbackErr)
```

Reply:
solved